Sources/MUXSDKStatsInternal/Publishers+AVPlayerItem.swift

```diff
@@ -0,0 +1,315 @@
+            return true
+        } ?? false
+    }
+
+    func send(_ event: MUXSDKBaseEvent) throws(CancellationError) {
+        try withMutableState { $0.receive(event) }
+    }
+}
+
+@available(iOS 16, tvOS 16, *)
+extension AVPlayerItem.MuxEventSubscription {
+
+    func startObserving(playerItem: AVPlayerItem) {
+        var cancellables = [AnyCancellable]()
+
+        let asset = playerItem.asset
+
+        let durationFuture = asset.future(for: .duration)
+        let variantsFuture = (asset as? AVURLAsset)?.future(for: .variants)
+
+        let tracksPublisher = playerItem.publisher(for: \.tracks, options: [.initial])
+            .removeDuplicates { tracksA, tracksB in
+                tracksA.elementsEqual(tracksB) { trackA, trackB in
+                    trackA.assetTrack?.trackID == trackB.assetTrack?.trackID
+                }
+            }
+            .share()
+
+        let statusPublisher = playerItem.publisher(for: \.status, options: [.initial])
+            .removeDuplicates()
+            .share()
+
+        let validPresentationSizePublisher = playerItem.publisher(for: \.presentationSize, options: [.initial])
+            .combineLatest(statusPublisher)
+            .drop { (size, status) in
+                // nonzero size is always valid - this will likely precede .readyToPlay
+                size == .zero && status != .readyToPlay
+            }
+            .map { (size, status) in size }
+            .removeDuplicates()
+            .share()
+
+        let videoAssetTrackPublisher = tracksPublisher
+            // AVPlayerItemTrack.assetTrack is @MainActor isolated:
+            .receive(on: ImmediateIfOnMainQueueScheduler.shared)
+            // See docs on AVPlayerItem.tracks - this will be empty until loaded
+            .drop(while: \.isEmpty)
+            .map { tracks in
+                tracks.lazy
+                    .compactMap(\.assetTrack)
+                    .first { $0.mediaType == .video }
+            }
+            .removeDuplicates { $0?.trackID == $1?.trackID }
+            .share()
+
+        let naturalSizePublisher = videoAssetTrackPublisher
+            .map { assetTrack -> AnyPublisher<CGSize?, Never> in
+                guard let assetTrack else {
+                    return Just<CGSize?>(nil).eraseToAnyPublisher()
+                }
+
+                return assetTrack.future(for: .naturalSize)
+                    .map(CGSize?.some)
+                    .replaceError(with: nil)
+                    .eraseToAnyPublisher()
+            }
+            .switchToLatest()
+            .removeDuplicates()
+            .share()
+
+        let activeVariantPublisher = variantsFuture?
+            .replaceError(with: [])
+            .combineLatest(naturalSizePublisher)
+            .map { variants, naturalSize in
+                variants.first { $0.videoAttributes?.presentationSize == naturalSize }
```

Comment:
If a customer's rendition ladder has variants at different bitrates or codecs but at the same resolution, will this still be accurate?

Reply:
There's a limitation we can do without some really hairy access log matching logic. I'm adding add a check to make sure we at least don't update with the _wrong_ bitrate, and a basic look into the access log to match on the most recent bitrate requested.

But yes, `.first` really isn't a good choice here.